### ISSUE TITLE
Implement a community list or abusiveip db

### ISSUE THREAD

**Issue by JReming85** [devnulli/EvlWatcher#71]
Would be nice to have a community managed list of bad actors or importable bad ip's.

I used to make lists with fail2ban and share them on a git long ago, but now its easier to just have them auto report using one of fail2ban's reporting actions.

Just think it may be "nice to have"  a sharable list of undesirable ip's.  The only downside is how it would be managed as I dont want to blacklist anyone just for using a questionable network (i.e. ColoCrossing for example)

https://www.projecthoneypot.org/list_of_ips.php looks to be a something I think would benefit everyone, actual ips caught red handed.  I would also be more then happy to direct my fail2ban ssh, smtp, dovecot, apache/nginx, and asterisk reporting if it would help

**Comment by devnulli** [comment]
Hey Jeremy, 

Sorry for the late reply (Im a freshly baked father :heart:) 

Uh yea, community based lists. It is THE **evergreen** of topics, and yea, I thought a lot about that. Also, @NukeDev  had something like that in mind. Generally, it's a cool idea, but it is a very difficult topic, and I want to share my thoughts about that with you, so you can also take them into consideration .

I want to bring up some bullet points. These are the things we would need to tackle before we can _seriously_ follow this issue:

**- >>About technical prerequisites (a.k.a what Windows Firewall and its API can handle)<<**

Before we think about that, we must find how many IPs the Windows Firewall-Entries can actually handle, and adjust our usage of the Windows firewall API to be mass-ban-able (API wise and Firewall wise). _(subnote: we should do that anyway, since i have NO idea how many IPs a) the Windows Firewall/b) and its API/c) the way we use the API can take, or how it influences system performance._ 

Generally, I think if we use it right, **this should be not a big problem, but its a research effort** to take. I think it will be somewhere in reasonable performance (but nothing compared to fail2ban/iptables of course). So lets put this point out of the discussion for a while.

**- >>About using community managed lists in general<<**

The next thing is, that community managed lists can also be a vulnerability. Imagine, this community WILL tell your system, whose IPs to block.  So - if we do something like that, we need to consume a trusteworthy existing list instead of making our own list, where some clown/joker could blank out North America -for instance- from our servers :grin: .

**- Next: Shooting at sparrows with high-tec nuclear missiles**

Also in my opinion, yielding a global list of malicious attackers DOES really help, especially against dynamic IPS. But is also a huge effort (filtering against a lot of criteria with little probability) - but we should not forget that EvlWatcher wants to be a simple, fail2ban like, thing. 

**- About directly importing this into the EvlWatcherService**

Next thing is: EvlWatcher runs as LocalSystem (not LocalNetwork), assuring every sysadmin that it will never communicate over the network in any way. I think thats nice, and would like to keep it that way. 

- About importing these lists via the console.

Like automatic updates, also could we import these lists via the management console. That could be a way to tackle this issue, but I think we should still not do it, because we should do something different:

**CONCLUSION:**

Using global blacklists is good, but should not be done with EvlWatcher. It's a fubar effort, and would make a massive and complex system out of EvlWatcher. We should rather:

- a) not reinvent the wheel: find out if there's already a tool available that can put and update existing trustworthy global blacklists into the windows firewalls, and maybe handle local lists (those local lists would then be fed by Evlwatcher by an adapter)

- b) if there happens to be nothing like that, we should MAKE something like that, start a new project for that, but let EvlWatcher keep its effortless simplicity and fail2ban style

Whats your thoughts about that?

**Comment by JReming85** [comment]
Completely agree, I was also thinking of having it as its own exe so those that dont want to contribute/use wont and the original "no external internet communication" standard will be honored with EvlWatcher.

It would even work to have it do the program update as well, check github to new version, send command to EvlWatcher to stop and close while it updates and then restarts the service when done.

**Comment by devnulli** [comment]
ill leave the discussion part of the issue open for a while as a reminder, if you start anything, let me know.
